src/PaymentBundle/Controller/PaymentController.php

```diff
@@ -161,14 +211,14 @@ protected function getBasketFactory()
      */
     protected function getBasket()
     {
-        return $this->get('sonata.basket');
+        return $this->basket;
```

Comment:
`$this->basket` may be `null` here, so you might want to add the same check as in `getBasketFactory` (same below in `getPaymentHandler`)

Reply:
You're so right too quick last modification